entity-framework/efcore-and-ef6/porting/index.md

```diff
@@ -1,61 +1,71 @@
-    * If the primary key is set to a value, the state is set to unchanged.
-  * If the primary key is not database generated, the entity is put in the same state as the root.
+This documentation uses the terms _port_ and _upgrade_ to avoid confusion with the term [_migrations_](/ef/core/managing-schemas/migrations/) as a feature of EF Core. Migrations in EF Core are fundamentally different than [EF6 Code First migrations](/ef/ef6/modeling/code-first/migrations/). There is not a recommend approach to port your migrations history, so plan to start "fresh" in EF Core. You can maintain the codebase and data from your EF6 migrations. Apply your final migration in EF6, then create an initial migration in EF Core. You will be able to track history in EF Core moving forward.
 
-### Code First database initialization
+## Upgrade steps
 
-**EF6 has a significant amount of magic it performs around selecting the database connection and initializing the database. Some of these rules include:**
+The upgrade path has been split into several documents that are organized by the phase of your upgrade and the type of application.
 
-* If no configuration is performed, EF6 will select a database on SQL Express or LocalDb.
+### Determine your EF Core "flavor"
 
-* If a connection string with the same name as the context is in the applications `App/Web.config` file, this connection will be used.
+There are several approaches to how EF Core works with your domain model and database implementation. In general, most apps will follow one of these patterns and how you approach your port will depend on the application "flavor.
 
-* If the database does not exist, it is created.
+**Code as the source of truth** is an approach in which everything is modeled through code and classes, whether through data attributes, fluent configuration, or a combination of both. The database is initially generated based on the model defined in EF Core and further updates are typically handled through migrations.
 
-* If none of the tables from the model exist in the database, the schema for the current model is added to the database. If migrations are enabled, then they are used to create the database.
+The **Database as source of truth** approach involves reverse-engineering or scaffolding your code from the database. When schema changes are made, the code is either regenerated or updated to reflect the changes.
 
-* If the database exists and EF6 had previously created the schema, then the schema is checked for compatibility with the current model. An exception is thrown if the model has changed since the schema was created.
+Finally, a more common **Hybrid mapping** approach follows the philosophy that the code and database are managed separately, and EF Core is used to map between the two. This approach may or may not use migrations.
```

Comment:
I love using "database is the source of truth" and "code is the source of truth" here. The hybrid mapping approach is maybe a bit harder to grok. Maybe save it for a more advanced section later? (Just ideas.)